src/tensors/indexmanipulations.jl

```diff
@@ -434,18 +434,54 @@ See also [`transpose`](@ref), [`transpose!`](@ref), [`add_permute!`](@ref), [`ad
                                             p::Index2Tuple,
                                             α::Number,
                                             β::Number,
-                                            backend::AbstractBackend...)
+                                            backend...)
     transformer = treetransposer(tdst, tsrc, p)
     return add_transform!(tdst, tsrc, p, transformer, α, β, backend...)
 end
 
+# Implementation
+# --------------
+"""
+    add_transform!(C, A, pA, transformer, α, β, [backend], [allocator])
+
+Return the updated `C`, which is the result of adding `α * A` to `β * B`,
+permuting the data with `pA` while transforming the fusiontrees with `transformer`.
+"""
+function add_transform!(C::AbstractTensorMap, A::AbstractTensorMap, pA::Index2Tuple,
+                        transformer, α::Number, β::Number)
+    return add_transform!(C, A, pA, transformer, α, β, TO.DefaultBackend())
+end
+function add_transform!(C::AbstractTensorMap, A::AbstractTensorMap, pA::Index2Tuple,
+                        transformer, α::Number, β::Number, backend)
+    return add_transform!(C, A, pA, transformer, α, β, backend, TO.DefaultAllocator())
+end
+function add_transform!(C::AbstractTensorMap, A::AbstractTensorMap, pA::Index2Tuple,
+                        transformer, α::Number, β::Number, backend, allocator)
+    if backend isa TO.DefaultBackend
+        newbackend = TO.select_backend(add_transform!, C, A)
+        return add_transform!(C, A, pA, transformer, α, β, newbackend, allocator)
+    elseif backend isa TO.NoBackend # error for missing backend
+        TC = typeof(C)
+        TA = typeof(A)
+        throw(ArgumentError("No suitable backend found for `add_transform!` and tensor types $TC and $TA"))
+    else # error for unknown backend
+        TC = typeof(C)
+        TA = typeof(A)
+        throw(ArgumentError("Unknown backend $backend for `add_transform!` and tensor types $TC and $TA"))
+    end
+end
+function TO.select_backend(::typeof(add_transform!), C::AbstractTensorMap,
```

Comment:
Should this one be moved into backends.jl like the ones for `TO.tensoradd!` and friends.